hail/src/main/scala/is/hail/annotations/StagedRegionValueBuilder.scala

```diff
@@ -240,31 +244,22 @@ class StagedRegionValueBuilder private(val mb: MethodBuilder, val typ: PType, va
     Region.storeDouble(currentOffset, v)
   }
 
-  def allocateBinary(n: Code[Int]): Code[Long] = {
-    val boff = mb.newField[Long]
-    Code(
-      boff := PBinary.allocate(region, n),
-      Region.storeInt(boff, n),
-      ftype match {
-        case _: PBinary => startOffset := boff
-        case _ =>
-          Region.storeAddress(currentOffset, boff)
-      },
-      boff)
-  }
-
   def addBinary(bytes: Code[Array[Byte]]): Code[Unit] = {
     val b = mb.newField[Array[Byte]]
     val boff = mb.newField[Long]
+    val pt = currentPType()
+
+    assert(pt.isInstanceOf[PBinary])
```

Comment:
remove. If this assertion fails, the next line would fail as well and with a better error message.